### PR TITLE
DEP-990 When source is pushed, update metadata on cp

### PR DESCRIPTION
**Pull request by cwesdorp**
When source was pushed, cp was not informed with the metadata. We'll call back CP on container start as then the application becomes available and supports dry-deploy in this way.

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -261,12 +262,19 @@ def run_mx_build():
...
     zf = zipfile.ZipFile('/tmp/model.mda')
     try:
         zf.extractall(BUILD_DIR)
     finally:
         zf.close()
 
+    try:
+        with open(os.path.join(BUILD_DIR, '.sourcepush'), 'w') as f:
+            f.write('sourcepush')
+    except:
```
[the remark this answers is not in the document]
Comment (cwesdorp) [reply bin/compile]: Done

File: bin/compile [hunk 2]
```diff
@@ -261,12 +262,19 @@ def run_mx_build():
...
+        logger.warn('Could not write source push indicator.')
+
```
[the remark this answers is not in the document]
Comment (cwesdorp) [reply bin/compile]: Done

File: lib/instadeploy.py [hunk 3]
```diff
@@ -295,3 +295,4 @@ def send_metadata_to_cloudportal():
         }
         # We don't care about the response.
         requests.post(target_url, files=files, headers=headers)
+        logger.info("Pushed metadata back home")
```
[the remark this answers is not in the document]
Comment (cwesdorp) [reply lib/instadeploy.py]: Done

File: bin/compile.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        zf.extractall(BUILD_DIR)
+    finally:
+        zf.close()
+
+    try:
+        with open(os.path.join(BUILD_DIR, ".sourcepush"), "w") as f:
+            f.write("sourcepush")
+    except Exception as e:
+        logging.warn(
+            "Could not write source push indicator. {}".format(str(e))
```
Comment (joshp123) [inline bin/compile.py]: For logging it is preferable not to use the `str.format` methods, but instead to use the old-style string interpolation syntax (`%s`), but passing the arguments to be interpolated as extra arguments to the logging call.

i.e.:
```suggestion
            "Could not write source push indicator. %s, str(e)
```

File: bin/compile.py [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+import buildpackutil
+from m2ee.version import MXVersion
+
+BUILDPACK_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+BUILD_DIR = sys.argv[1]
+CACHE_DIR = os.path.join(sys.argv[2], "bust")
+DOT_LOCAL_LOCATION = os.path.join(BUILD_DIR, ".local")
+BUILD_ERRORS_JSON = "/tmp/builderrors.json"
+
+# sys.path.insert(0, os.path.join(BUILDPACK_DIR, 'lib'))
```
Comment (joshp123) [inline bin/compile.py]: Please remove the commented out code.

File: bin/compile.py [hunk 6]
```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python3
+import codecs
```
Comment (joshp123) [inline bin/compile.py]: Please follow PEP8 for imports: https://www.python.org/dev/peps/pep-0008/#imports

Specifically, the following:

>Imports should be grouped in the following order:
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.
>
>You should put a blank line between each group of imports.

**Pull request merged**